docs/modules/tiered-storage/pages/overview.adoc

```diff
@@ -1,37 +1,85 @@
+See xref:deploy:enterprise-licenses.adoc[].
 
-Tiered Storage uses the following:
+== High-Level Design Overview
 
-* An in-memory index to allow faster access to map entries.
-* A special memory manager to decide which entries to store in the memory tier or the disk tier. If a map entry is not accessed for some time, it gets flushed to the disk. If a map is frequently accessed, it stays in memory.
+Tiered Storage for IMaps is backed by a special-purpose, addressable data structure called HybridLog.
+It is called HybridLog because it seamlessly combines in-place updates with a traditional append-only log.
+This data structure spans the memory and the configured disk to provide the aggregated capacity of the resources available for storing IMap's entries.
+There is a HybridLog data structure for every partition of every IMap.
 
-== Not Yet Supported
+The IMap entries stored in these HybridLog instances are internally referenced by their logical addresses.
+The key to logical address mapping uses a hash index component.
+These hash indices are hashtables using separate chaining for conflict resolution.
+The table of these hash indices is stored in native memory; the chaining information for the buckets is stored on the disk.
 
-- Data recovery from persisted data: When a Hazelcast member has Tiered Storage enabled and it fails or shuts down, you are not able to use the local device files of Tiered Storage for data recovery. Data recovery will be supported in the future releases.
+Tiered Storage ensures that every accessed IMap entry is available in the memory to build a hot cache of entries.
+This means that, if an entry is on the disk, it is relocated to the memory inside the HybridLog hosting the entry, and its previous location on the disk becomes unreachable.
+As more and more entries are relocated to the in-memory region of the HybridLog, the memory space occupied by the unreachable entries on disk is reclaimed.
+This reclamation is done by a compactor running synchronously with IMap operations on the same partition.
 
-- Time-to-live (TTL) expiration is not supported by the Tiered-Storage-backed maps. This means that methods, such as `IMap.put`, throw `UnsupportedOperationException` if a TTL value is provided. If the default TTL setting is present in the map configuration, `InvalidConfigurationException` is thrown during the member startup.
+== Example Use Case
 
-- Max-Idle expiration is not supported by the Tiered-Storage-backed maps. This means that methods, such as `IMap.put`, throw `UnsupportedOperationException` if a Max-Idle value is provided. If the default Max-Idle setting is present in the map configuration, `InvalidConfigurationException` is thrown during the member startup.
+Efficiently managing transactional data is crucial in banking applications. Traditional setups like involving memory and a database can be complex and costly to maintain.
```

Comment:
I'm afraid this statement may put a target on us. It might give the reader the impression that tiered storage deals with data with strong consistency guarantees, while it's not the case. What's [written](https://docs.hazelcast.com/hazelcast/5.3/data-structures/map#availability-vs-consistency) for distributed maps still holds. I'd either reference that page, or make this remark in any way, or remove this example.